ciao/cmd/root.go

```diff
@@ -42,6 +43,25 @@ func render(cmd *cobra.Command, data interface{}) error {
 		"Error generating template output")
 }
 
+func templatedUsageFunc(cmd *cobra.Command) error {
+	err := rootUsageFunc(cmd)
+	if err != nil {
+		return err
+	}
+
+	if cmd.Annotations == nil {
+		return nil
+	}
+
+	templateUsage := cmd.Annotations["template_usage"]
+	if templateUsage != "" {
+		fmt.Printf("When using the template flag the following structure is provided:\n\n")
```

Comment:
If might be nice to prepend a '\n' here.  This is what I currently see.

```
$ ciao list cncis
Error: Listing CNCIs is limited to privileged users
When using the template flag the following structure is provided:

[]struct {
```
and I'm thinking

```
$ ciao list cncis
Error: Listing CNCIs is limited to privileged users

When using the template flag the following structure is provided:

[]struct {
```

would be a bit nicer.

Reply:
done